pkg/aws/eni/metrics/metrics.go

```diff
@@ -67,13 +68,20 @@ func NewPrometheusMetrics(namespace string, registry *prometheus.Registry) *prom
 		Help:      "Number of ENIs allocated",
 	}, []string{"subnetId", "status"})
 
-	m.Available = prometheus.NewGauge(prometheus.GaugeOpts{
+	m.AvailableENIs = prometheus.NewGauge(prometheus.GaugeOpts{
 		Namespace: namespace,
 		Subsystem: eniSubsystem,
 		Name:      "available",
 		Help:      "Number of ENIs with addresses available",
 	})
 
+	m.AvailableIPsPerSubnet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: namespace,
+		Subsystem: eniSubsystem,
+		Name:      "available_ips_per_subnet",
+		Help:      "Number of available IPs per subnet ID",
+	}, []string{"subnetId", "availabilityZone"})
```

Comment:
Interested in what you folks think would be useful here as well. From an UX perspective having the `subnetId` is pretty weird because they are generated by AWS. AZ helps a bit but does not give a lot of insight if there are multiple subnets in the same zone. 

Depending on the VPC setup you might know in what AZs are the subnets that keep pod ips. I'm thinking to add the subnet tags as labels as well. I don't imagine this to add a lot of cardinality to these metrics.

Reply:
From a cardinality perspective, subnetId should still be fine. From a UX perspective, the user can choose whether to include or exclude the subnetId. It might be helpful to track a particular subnet but it's likely not a good default graph setting.

Reply:
I did a bit more research here and did not find a great way to inject labels on the fly on prometheus `GaugeVec` (although my experience with prometheus is pretty limited). Let's drop this for the moment.